www/docs/intro.md

```diff
@@ -375,6 +376,43 @@ Devnet offers [empty block creation](https://shard-labs.github.io/starknet-devne
 const emptyBlock = await starknet.devnet.createBlock();
 ```
 
+## Debugging contracts
+
+To debug StarkNet contracts, you can use `print()` in cairo hints in your contract, and the printed lines will appear in Devnet's log.
+
+Compile with `--disable-hint-validation` flag to allow hints.
+
+```
+hardhat starknet-compile --disable-hint-validation
+```
+
+For example, when calling the following `increase_balance` with input `25`.
+
+```cairo
+@external
+func increase_balance{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    amount: felt
+) {
+    let (res) = balance.read();
+    %{ print( "Adding balance..." ) %}
+    %{ print( ids.res ) %}
+    balance.write(res + amount);
+    let (afterUpdate) = balance.read();
+    %{ print( ids.afterUpdate ) %}
+    return ();
+}
+```
+
+DevNet logs look like this,
+
+```
+Adding balance...
+0
+25
+```
+
+When using `integrated-devnet` you can specify `stdout` as `STDOUT` or a file in your hardhat config file. Documented in [runtime network](#runtime-network---integrated-devnet) section.
```

Comment:
Here I would motivate the adoption of integrated-devnet by saying:
```
If you want to have your debug lines printed in the same terminal as your hardhat script/test, use integrated-devnet with `stdout` set to `"STDOUT"` as documented in ...
```

Reply:
Addressed in dd2e5e617cd5241599ab5bde9729ebaea8252f1b